base_address_parish/models/res_parish.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2022 Riverminds Cia Ltda - Mamfredy Mejia Matute
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, exceptions, fields, models
+
+
+class ResParish(models.Model):
+    _name = "res.parish"
+    _order = "name"
+
+    name = fields.Char("Name", required=True, translate=True)
+    code = fields.Char("Code")
+    country_id = fields.Many2one('res.country', string='Country', required=True)
+    state_id = fields.Many2one(
+        'res.country.state', 'State', domain="[('country_id', '=', country_id)]")
+    city_id = fields.Many2one(
```

Comment:
If parish can belong to city, why you should specific country or state?, this fields(country_id, state_id) can be related with res.city fields, with this design you can set different country for a city, or state